Build failure: efitools-1.9.2

## Steps To Reproduce

Steps to reproduce the behavior:

1. build efitools-1.9.2

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/19qizmg3xsr8x1gw7qsxh8gbbxvinh5z-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/w4vbi7zysl1gqfsk2fiml8677bkpxvf9-aarch64.patch
patching file Make.rules
substituteStream() in derivation efitools-1.9.2: WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file 'lib/console.c')
patching script interpreter paths in .
./mkusb.sh: interpreter directive changed from "#!/bin/sh" to "/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/sh"
./xxdi.pl: interpreter directive changed from "#!/usr/bin/env perl" to "/nix/store/1i003ijlh9i0mzp6alqby5hg3090pjdx-perl-5.40.0/bin/perl"
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: SHELL=/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash
make -C lib lib-efi.a
make[1]: Entering directory '/build/source/lib'
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -fno-toplevel-reorder -DBUILD_EFI -c simple_file.c -o simple_file.efi.o
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -fno-toplevel-reorder -DBUILD_EFI -c pecoff.c -o pecoff.efi.o
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -fno-toplevel-reorder -DBUILD_EFI -c guid.c -o guid.efi.o
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -fno-toplevel-reorder -DBUILD_EFI -c sha256.c -o sha256.efi.o
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -fno-toplevel-reorder -DBUILD_EFI -c console.c -o console.efi.o
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -fno-toplevel-reorder -DBUILD_EFI -c execute.c -o execute.efi.o
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -fno-toplevel-reorder -DBUILD_EFI -c configtable.c -o configtable.efi.o
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -fno-toplevel-reorder -DBUILD_EFI -c shell.c -o shell.efi.o
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -fno-toplevel-reorder -DBUILD_EFI -c security_policy.c -o security_policy.efi.o
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -fno-toplevel-reorder -DBUILD_EFI -c shim_protocol.c -o shim_protocol.efi.o
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -fno-toplevel-reorder -DBUILD_EFI -c pkcs7verify.c -o pkcs7verify.efi.o
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -c variables.c -o variables.o
ar rcv lib-efi.a simple_file.efi.o pecoff.efi.o guid.efi.o sha256.efi.o console.efi.o execute.efi.o configtable.efi.o shell.efi.o security_policy.efi.o shim_protocol.efi.o pkcs7verify.efi.o variables.o
a - simple_file.efi.o
a - pecoff.efi.o
a - guid.efi.o
a - sha256.efi.o
a - console.efi.o
a - execute.efi.o
a - configtable.efi.o
a - shell.efi.o
a - security_policy.efi.o
a - shim_protocol.efi.o
a - pkcs7verify.efi.o
a - variables.o
make[1]: Leaving directory '/build/source/lib'
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -c HelloWorld.c -o HelloWorld.o
ld -nostdlib -shared -Bsymbolic /nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/lib//crt0-efi-x86_64.o -L /nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/lib/ -L /usr/lib -L /usr/lib64 -T elf_x86_64_efi.lds HelloWorld.o lib/lib-efi.a -o HelloWorld.so -lefi -lgnuefi /nix/store/l89iqc7am6i60y8vk507zwrzxf0wcd3v-gcc-14-20241116/lib/gcc/x86_64-unknown-linux-gnu/14.2.1/libgcc.a
# check we have no undefined symbols
nm -D HelloWorld.so | grep ' U ' && exit 1 || exit 0
objcopy -j .text -j .sdata -j .data -j .dynamic -j .dynsym \
	   -j .rel -j .rela -j .rel.* -j .rela.* -j .rel* -j .rela* \
	   -j .reloc --target=efi-app-x86_64 HelloWorld.so HelloWorld.efi
openssl req -new -x509 -newkey rsa:2048 -subj "/CN=DB/" -keyout DB.key -out DB.crt -days 3650 -nodes -sha256
....+......+++++++++++++++++++++++++++++++++++++++*......+...+++++++++++++++++++++++++++++++++++++++*.....+.....+....+......+.....+...+...+.......+..+....+..+...+........................+......+.+........+...+...+.+..............+.+..............+.+.....+.+.....+.......+...+.....+.......+..+..........+.....+.........+.......+...+..+.......+..+.........+.+.........+..+....+...+...+........+....+...+.....+...+.........+.+.........+..+......+...+.+...+...+.....+...+.........+.............+..+.+.....+.+...+..+.+..+............+.+...............+.........+.....+.........+.++++++
...+.+...+++++++++++++++++++++++++++++++++++++++*.+++++++++++++++++++++++++++++++++++++++*......+...+.......+..................+.....+...............+.............++++++
-----
sbsign --key DB.key --cert DB.crt --output HelloWorld-signed.efi HelloWorld.efi
warning: data remaining[43008 vs 52751]: gaps between PE/COFF sections?
warning: data remaining[43008 vs 52752]: gaps between PE/COFF sections?
Signing Unsigned original image
openssl req -new -x509 -newkey rsa:2048 -subj "/CN=PK/" -keyout PK.key -out PK.crt -days 3650 -nodes -sha256
..+.....+......+....+......+......+.....+...+..........+........+...+++++++++++++++++++++++++++++++++++++++*..+........+...+......+..........+......+...+......+++++++++++++++++++++++++++++++++++++++*......+.....+.............+...+..+.+..+.........+.+...+......+.......................+.............+...+..+.......+.....+...+.........+....+.....+....+........+....+........+.......+.....+.............+...+.....+...+..........+..+.......+..+...+.+......+...+..+....+.....+..........+..+...................+.....+......+...+....+..+.+............+...+.....+..........+..+............+...+.+..+....+......+...+..+...+.+...+...+........+......+.+...+.........+..+............................+..+.............+.....+...............+.......+...+..+.+..+...............+.+.....+....+.....+.............+............+...+.....+.........++++++
....+++++++++++++++++++++++++++++++++++++++*............+++++++++++++++++++++++++++++++++++++++*...+..+.........+...+.+............+..+.+..+...+......+...+.......+.........+.....+.+........+..........+...+...+.....+....+.....+.+.........+..+................+..............+.+.........+......+..+...+..................+.......+...+...........+...+...+............+...+...+....+.....+.+.........+..+..........+.........+............+..+......+.+...+........+.........+.+..+...+............+......+.+..............+.+...+...+...+.................+..........+..+............+.+...+...........+.+.....+.........+...+.+..+.......++++++
-----
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -c cert-to-efi-sig-list.c -o cert-to-efi-sig-list.o
[01m[Kcert-to-efi-sig-list.c:9:9:[m[K [01;35m[Kwarning: [m[K"__STDC_VERSION__" redefined
    9 | #define [01;35m[K__STDC_VERSION__[m[K 199901L
      |         [01;35m[K^~~~~~~~~~~~~~~~[m[K
[01m[K<built-in>:[m[K [01;36m[Knote: [m[Kthis is the location of the previous definition
make -C lib lib.a
make[1]: Entering directory '/build/source/lib'
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -c simple_file.c -o simple_file.o
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -c pecoff.c -o pecoff.o
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -c guid.c -o guid.o
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -c sha256.c -o sha256.o
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -c console.c -o console.o
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -c execute.c -o execute.o
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -c configtable.c -o configtable.o
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -c shell.c -o shell.o
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -c security_policy.c -o security_policy.o
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -c shim_protocol.c -o shim_protocol.o
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -c pkcs7verify.c -o pkcs7verify.o
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -c kernel_efivars.c -o kernel_efivars.o
[01m[Kkernel_efivars.c:19:9:[m[K [01;35m[Kwarning: [m[K"__STDC_VERSION__" redefined
   19 | #define [01;35m[K__STDC_VERSION__[m[K 199901L
      |         [01;35m[K^~~~~~~~~~~~~~~~[m[K
[01m[K<built-in>:[m[K [01;36m[Knote: [m[Kthis is the location of the previous definition
[01m[Kkernel_efivars.c:[m[K In function '[01m[Kkernel_variable_init[m[K':
[01m[Kkernel_efivars.c:61:9:[m[K [01;35m[Kwarning: [m[Kignoring return value of '[01m[Kread[m[K' declared with attribute '[01m[Kwarn_unused_result[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-unused-result-Wunused-result]8;;[m[K]
   61 |         [01;35m[Kread(fd, buf, st.st_size)[m[K;
      |         [01;35m[K^~~~~~~~~~~~~~~~~~~~~~~~~[m[K
[01m[Kkernel_efivars.c:[m[K In function '[01m[Kget_variable[m[K':
[01m[Kkernel_efivars.c:107:9:[m[K [01;35m[Kwarning: [m[Kignoring return value of '[01m[Kread[m[K' declared with attribute '[01m[Kwarn_unused_result[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-unused-result-Wunused-result]8;;[m[K]
  107 |         [01;35m[Kread(fd, &attr, sizeof(attr))[m[K;
      |         [01;35m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
[01m[Kkernel_efivars.c:113:17:[m[K [01;35m[Kwarning: [m[Kignoring return value of '[01m[Kread[m[K' declared with attribute '[01m[Kwarn_unused_result[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-unused-result-Wunused-result]8;;[m[K]
  113 |                 [01;35m[Kread(fd, buf, st.st_size - sizeof(attr))[m[K;
      |                 [01;35m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -c openssl_sign.c -o openssl_sign.o
[01m[Kopenssl_sign.c:[m[K In function '[01m[Kread_engine_private_key[m[K':
[01m[Kopenssl_sign.c:118:9:[m[K [01;35m[Kwarning: [m[K'[01m[KENGINE_load_builtin_engines[m[K' is deprecated: Since OpenSSL 3.0 [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdeprecated-declarations-Wdeprecated-declarations]8;;[m[K]
  118 |         [01;35m[KENGINE_load_builtin_engines[m[K();
      |         [01;35m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
In file included from [01m[Kopenssl_sign.c:10[m[K:
[01m[K/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/engine.h:358:28:[m[K [01;36m[Knote: [m[Kdeclared here
  358 | OSSL_DEPRECATEDIN_3_0 void [01;36m[KENGINE_load_builtin_engines[m[K(void);
      |                            [01;36m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
[01m[Kopenssl_sign.c:119:9:[m[K [01;35m[Kwarning: [m[K'[01m[KENGINE_by_id[m[K' is deprecated: Since OpenSSL 3.0 [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdeprecated-declarations-Wdeprecated-declarations]8;;[m[K]
  119 |         [01;35m[Ke[m[K = ENGINE_by_id(engine);
      |         [01;35m[K^[m[K
[01m[K/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/engine.h:336:31:[m[K [01;36m[Knote: [m[Kdeclared here
  336 | OSSL_DEPRECATEDIN_3_0 ENGINE *[01;36m[KENGINE_by_id[m[K(const char *id);
      |                               [01;36m[K^~~~~~~~~~~~[m[K
[01m[Kopenssl_sign.c:135:9:[m[K [01;35m[Kwarning: [m[K'[01m[KENGINE_init[m[K' is deprecated: Since OpenSSL 3.0 [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdeprecated-declarations-Wdeprecated-declarations]8;;[m[K]
  135 |         [01;35m[Kif[m[K (!ENGINE_init(e)) {
      |         [01;35m[K^~[m[K
[01m[K/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/engine.h:620:27:[m[K [01;36m[Knote: [m[Kdeclared here
  620 | OSSL_DEPRECATEDIN_3_0 int [01;36m[KENGINE_init[m[K(ENGINE *e);
      |                           [01;36m[K^~~~~~~~~~~[m[K
[01m[Kopenssl_sign.c:141:9:[m[K [01;35m[Kwarning: [m[K'[01m[KENGINE_load_private_key[m[K' is deprecated: Since OpenSSL 3.0 [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdeprecated-declarations-Wdeprecated-declarations]8;;[m[K]
  141 |         [01;35m[Kpkey[m[K = ENGINE_load_private_key(e, keyfile, ui, NULL);
      |         [01;35m[K^~~~[m[K
[01m[K/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/engine.h:638:11:[m[K [01;36m[Knote: [m[Kdeclared here
  638 | EVP_PKEY *[01;36m[KENGINE_load_private_key[m[K(ENGINE *e, const char *key_id,
      |           [01;36m[K^~~~~~~~~~~~~~~~~~~~~~~[m[K
[01m[Kopenssl_sign.c:142:9:[m[K [01;35m[Kwarning: [m[K'[01m[KENGINE_finish[m[K' is deprecated: Since OpenSSL 3.0 [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdeprecated-declarations-Wdeprecated-declarations]8;;[m[K]
  142 |         [01;35m[KENGINE_finish[m[K(e);
      |         [01;35m[K^~~~~~~~~~~~~[m[K
[01m[K/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/engine.h:628:27:[m[K [01;36m[Knote: [m[Kdeclared here
  628 | OSSL_DEPRECATEDIN_3_0 int [01;36m[KENGINE_finish[m[K(ENGINE *e);
      |                           [01;36m[K^~~~~~~~~~~~~[m[K
[01m[Kopenssl_sign.c:145:9:[m[K [01;35m[Kwarning: [m[K'[01m[KENGINE_free[m[K' is deprecated: Since OpenSSL 3.0 [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdeprecated-declarations-Wdeprecated-declarations]8;;[m[K]
  145 |         [01;35m[KENGINE_free[m[K(e);
      |         [01;35m[K^~~~~~~~~~~[m[K
[01m[K/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/engine.h:493:27:[m[K [01;36m[Knote: [m[Kdeclared here
  493 | OSSL_DEPRECATEDIN_3_0 int [01;36m[KENGINE_free[m[K(ENGINE *e);
      |                           [01;36m[K^~~~~~~~~~~[m[K
ar rcv lib.a simple_file.o pecoff.o guid.o sha256.o console.o execute.o configtable.o shell.o security_policy.o shim_protocol.o pkcs7verify.o kernel_efivars.o openssl_sign.o
a - simple_file.o
a - pecoff.o
a - guid.o
a - sha256.o
a - console.o
a - execute.o
a - configtable.o
a - shell.o
a - security_policy.o
a - shim_protocol.o
a - pkcs7verify.o
a - kernel_efivars.o
a - openssl_sign.o
make[1]: Leaving directory '/build/source/lib'
gcc  -o cert-to-efi-sig-list cert-to-efi-sig-list.o lib/lib.a -lcrypto
./cert-to-efi-sig-list -g 11111111-2222-3333-4444-123456789abc PK.crt PK.esl
openssl req -new -x509 -newkey rsa:2048 -subj "/CN=KEK/" -keyout KEK.key -out KEK.crt -days 3650 -nodes -sha256
..........+.+......+..............+...+.......+...+..+......+++++++++++++++++++++++++++++++++++++++*.+...+..........+..+...+................+........+.+..+....+...+......+..+.......+..+..........+.....+...+......+....+............+.....+......+..........+..+.+......+++++++++++++++++++++++++++++++++++++++*..+.......+.....+...+............+......+.......+.....+.......+...+.....+.+..+...+.+.........+..+.......+.....+.......+..............+.......+..+...+...+..........+..+.......+..+.............+....................+......+.......+..+............+.+..+..................+...+....+...+........+.........+...+...............+....+......+...+......++++++
.....+..+.+...........+....+......+...+...+........+.+.....+................+...+..+.......+..+......+...+++++++++++++++++++++++++++++++++++++++*....+...............+......+......+....+..+.......+..+.+.....+...+.+.....+..................+...+....+.....+....+......+...+...+.....+....+.........+............+..+.+..........................+.............+..+.+...+......+...+..+...+.+..+++++++++++++++++++++++++++++++++++++++*..+...+.....+.+............+.....+......................+...........+......+....+..+.........+....+............+..+.+..............+...............+.+......+......+..+......+.+..............+......+...+.......+...+......+............+.....+.......+..+.+........+.+.....+...+.+.....+.......+.....+....+...+......+..+...+....+..+..........+.....+.........+.+...+........+...+.+...+.....................+......+......+.........+...+......+..+.........+.+.........+.........+.....+.+......+...+..+............+.+.....+.+........+.+..+...+....+........+...............+.............+......+...+..+......+...............+.......+.....+.+...............+.........+..+...+...+.............+..+.......+...+.........+.........+..+...+.........+......+......+.+.........+...........+..........+........+......+.........+............+.......+..+...+.+.....+.+......+.....+...+.........+.+......+.....+.+.........+...+...........+.......+..............+.+............++++++
-----
gcc -I/build/source/include/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi/ -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//x86_64 -I/nix/store/503ajjvv79z7wbmp41xjk9q8bkijjjc4-gnu-efi-3.0.18/include/efi//protocol -O2 -g  -fpic -Wall -fshort-wchar -fno-strict-aliasing -fno-merge-constants -fno-stack-protector -ffreestanding -fno-stack-check -DGNU_EFI_USE_MS_ABI -DEFI_FUNCTION_WRAPPER -mno-red-zone -DCONFIG_x86_64 -c sign-efi-sig-list.c -o sign-efi-sig-list.o
[01m[Ksign-efi-sig-list.c:7:9:[m[K [01;35m[Kwarning: [m[K"__STDC_VERSION__" redefined
    7 | #define [01;35m[K__STDC_VERSION__[m[K 199901L
      |         [01;35m[K^~~~~~~~~~~~~~~~[m[K
[01m[K<built-in>:[m[K [01;36m[Knote: [m[Kthis is the location of the previous definition
[01m[Ksign-efi-sig-list.c:[m[K In function '[01m[Kmain[m[K':
[01m[Ksign-efi-sig-list.c:166:17:[m[K [01;31m[Kerror: [m[Kimplicit declaration of function '[01m[Kstrptime[m[K'; did you mean '[01m[Kstrftime[m[K'? [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wimplicit-function-declaration-Wimplicit-function-declaration]8;;[m[K]
  166 |                 [01;31m[Kstrptime[m[K(timestampstr, "%Y-%m-%d %H:%M:%S", &tms);
      |                 [01;31m[K^~~~~~~~[m[K
      |                 [32m[Kstrftime[m[K
[01m[Ksign-efi-sig-list.c:224:9:[m[K [01;35m[Kwarning: [m[Kignoring return value of '[01m[Kread[m[K' declared with attribute '[01m[Kwarn_unused_result[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-unused-result-Wunused-result]8;;[m[K]
  224 |         [01;35m[Kread(fdefifile, ptr, st.st_size)[m[K;
      |         [01;35m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
[01m[Ksign-efi-sig-list.c:245:17:[m[K [01;35m[Kwarning: [m[Kignoring return value of '[01m[Kread[m[K' declared with attribute '[01m[Kwarn_unused_result[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-unused-result-Wunused-result]8;;[m[K]
  245 |                 [01;35m[Kread(infile, sigbuf, sigsize)[m[K;
      |                 [01;35m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
[01m[Ksign-efi-sig-list.c:283:9:[m[K [01;35m[Kwarning: [m[Kignoring return value of '[01m[Kwrite[m[K' declared with attribute '[01m[Kwarn_unused_result[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-unused-result-Wunused-result]8;;[m[K]
  283 |         [01;35m[Kwrite(fdoutfile, out, outlen)[m[K;
      |         [01;35m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
[01m[Ksign-efi-sig-list.c:286:17:[m[K [01;35m[Kwarning: [m[Kignoring return value of '[01m[Kwrite[m[K' declared with attribute '[01m[Kwarn_unused_result[m[K' [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-unused-result-Wunused-result]8;;[m[K]
  286 |                 [01;35m[Kwrite(fdoutfile, ptr, st.st_size)[m[K;
      |                 [01;35m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
make: *** [Make.rules:97: sign-efi-sig-list.o] Error 1
rm HelloWorld.o

```

</details>

## Additional context

<!-- Add any other context about the problem here. -->

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ef/efitools/package.nix
+++ b/pkgs/by-name/ef/efitools/package.nix
@@ -39,6 +39,7 @@ stdenv.mkDerivation rec {
     sed -i -e 's#/usr/include/efi#${gnu-efi}/include/efi/#g' Make.rules
     sed -i -e 's#/usr/lib64/gnuefi#${gnu-efi}/lib/#g' Make.rules
     sed -i -e 's#$(DESTDIR)/usr#$(out)#g' Make.rules
+    sed -i '$asign-efi-sig-list.o flash-var.o: CFLAGS += -D_GNU_SOURCE' Makefile
     substituteInPlace lib/console.c --replace "EFI_WARN_UNKOWN_GLYPH" "EFI_WARN_UNKNOWN_GLYPH"
     patchShebangs .
   '';

